### PR TITLE
Delete SSM secrets on cluster deletion

### PR DESCRIPTION
**Pull request by ivaruf**
## Description
Removes secrets (SSM parameters) when deleting cluster

## Motivation and Context
Cleanup resources on cluster deletion

## How Has This Been Tested?
Manually

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I have updated the release notes (for the next release).

### CODE REVIEW

File: pkg/client/core/service_parameter.go [hunk 1]
```diff
@@ -16,6 +21,33 @@ type parameterService struct {
...
+	err := s.spinner.Start("parameter")
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		err = s.spinner.Stop()
+	}()
+
+	fmt.Println(name)
```
Comment (ivaruf) [inline pkg/client/core/service_parameter.go:34]: Intentional printline - considering sending (o.Err down here) or something else, even though this should be refactored as soon as possible.

File: cmd/okctl/delete.go [hunk 2]
```diff
@@ -131,6 +131,39 @@ including VPC, this is a highly destructive operation.`,
...
+			if argoCdSecretPath != "" {
+				err = services.Parameter.DeleteSecret(o.Ctx, o.CloudProvider, argoCdSecretPath)
+				if err != nil {
+					return formatErr(err)
+				}
+			}
+
+			for c := range clients {
+				clientSecretPath := clients[c].ClientSecret.Path
+				if clientSecretPath != "" {
```
Comment (yngvark) [inline cmd/okctl/delete.go:150]: Do we know that ClientSecret isn't nil? Just checking.

File: cmd/okctl/delete.go [hunk 3]
```diff
@@ -131,6 +131,39 @@ including VPC, this is a highly destructive operation.`,
...
+					err := services.Parameter.DeleteSecret(o.Ctx, o.CloudProvider, clientSecretPath)
+					if err != nil {
+						return formatErr(err)
+					}
+				}
+			}
+
+			for k := range repos {
+				repoSecretPath := repos[k].DeployKey.PrivateKeySecret.Path
```
Comment (yngvark) [inline cmd/okctl/delete.go:159]: Do we know that DeployKey and PRivateKeySecret isn't nil? Just checking.

File: pkg/api/core/service_parameter.go [hunk 4]
```diff
@@ -5,13 +5,20 @@ import (
...
 
 type parameter struct {
 	cloudProvider api.ParameterCloudProvider
 	store         api.ParameterStore
 }
 
+// nolint: godox
+// TODO, implement this in proper client / api fashion.
+func (p *parameter) DeleteSecret(ctx context.Context, provider v1alpha1.CloudProvider, name string) error {
+	panic("Unused on the api side. Pay of this tech debt.")
```
Comment (yngvark) [inline pkg/api/core/service_parameter.go:19]: of -> off 

😄

**Review by yngvark — APPROVED** [review]

**Review by yngvark — COMMENTED** [review]
https://user-images.githubusercontent.com/562343/102778932-b6bc9d80-4393-11eb-86fa-55785723e358.mp4

(👆 Need to test the new mp4 support from github.)

File: cmd/okctl/delete.go (hunk 2 above)
Comment (ivaruf) [reply cmd/okctl/delete.go:150]: Have tested it this time ... , if it is not found it will be emptystring

File: cmd/okctl/delete.go (hunk 3 above)
Comment (ivaruf) [reply cmd/okctl/delete.go:159]: (same answer :) )

**Pull request merged**